tests/foreman/ui/test_contentview.py

```diff
@@ -1830,115 +1834,406 @@ def test_positive_restart_publish_via_dynflow(self):
-        # but these tests require creating a user with admin permissions
-        # for Content Views
-        # Dev note: none of this stuff is integrated with foreman rbac yet
-        # As such, all variations in here subject to change.
-        # Variations:
-        #  * Read, Modify, Delete, Promote Publish, Subscribe
-        """attempt to view content views
+        """Attempt to manage content views
 
         @id: c4d270fc-a3e6-4ae2-a338-41d864a5622a
 
-        @setup: create a user with the Content View admin role
+        @steps:
+        with global admin account:
 
-        @assert: User with admin role for content view can perform all
-        Variations above
+        1. create a user with all content views permissions
+        2. create lifecycle environment
+        3. create 2 content views (one to delete, the other to manage)
 
-        @caseautomation: notautomated
+        @setup: create a user with all content views permissions
 
+        @assert: The user can Read, Modify, Delete, Publish, Promote the
+        content views
 
         @CaseLevel: Integration
         """
+        # note: the user to create has no permissions to access products
+        # repositories
+        repo_name = gen_string('alpha')
+        cv_name = gen_string('alpha')
+        cv_new_name = gen_string('alpha')
+        cv_copy_name = gen_string('alpha')
+        env_name = gen_string('alpha')
+        user_login = gen_string('alpha')
+        user_password = gen_string('alphanumeric')
+        role_name = gen_string('alpha')
+        # create a role with all content views permissions
+        role = entities.Role(name=role_name).create()
+        entities.Filter(
+            organization=[self.organization],
+            permission=entities.Permission(
+                resource_type='Katello::ContentView').search(),
+            role=role,
+            search=None
+        ).create()
+        # create environment permissions with read only and promote access
+        # to content views
+        env_permissions_entities = entities.Permission(
+            resource_type='Katello::KTEnvironment').search()
+        user_env_permissions = [
+            'promote_or_remove_content_views_to_environments',
+            'view_lifecycle_environments'
+        ]
+        user_env_permissions_entities = [
+            entity
+            for entity in env_permissions_entities
+            if entity.name in user_env_permissions
+        ]
+        entities.Filter(
+            organization=[self.organization],
+            permission=user_env_permissions_entities,
+            role=role,
+            # allow access only to the mentioned here environments
+            search='name = {0} or name = {1}'.format(ENVIRONMENT, env_name)
+        ).create()
+        # create a user and assign the above created role
+        entities.User(
+            default_organization=self.organization,
+            organization=[self.organization],
+            role=[role],
```

Comment:
Yes right, otherwise if the user has not these additional 2 LCE permissions, he cannot  publish and promote the content view. So an admin user for content views should have not only permissions for content views but also some LCE permissions.